arrow/locales.py

```diff
@@ -2755,11 +2755,13 @@ class MoroccoArabicLocale(ArabicLocale):
 class IcelandicLocale(Locale):
     def _format_timeframe(self, timeframe: TimeFrameLiteral, delta: int) -> str:
         form = self.timeframes[timeframe]
-        if delta < 0:
-            form = form[0]
-        elif delta > 0:
-            form = form[1]
-            # FIXME: handle when delta is 0
+
+        if isinstance(form, Mapping):
+            if delta < 0:
+                form = form["past"]
+            elif delta >= 0:
+                form = form["future"]
+                # FIXME: handle when delta is 0
```

Comment:
I guess this is now fixed when `delta=0`?

Reply:
So I think there are two options here. We default to the future value when the case is that the delta for a particular unit is zero, or we implicitly error out like we currently do. In an ideal scenario, I would add a raise value error here, but I fear that would probably break the use of the locale, given how common the use of a delta of zero is.

Reply:
I would tend to think error out is better as it is probably less surprising.

Reply:
```shell
(arrow) chris@Z490:~/arrow$ python
Python 3.9.1 (default, Feb 17 2021, 16:56:42) 
[GCC 10.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import arrow
>>> start=arrow.utcnow()
>>> start.humanize(other=start, locale="is")
'rétt í þessu'
>>> start.humanize(other=start, granularity="hour",locale="is")
'eftir 0 tíma'
```